lib/util/processErrors.ts

```diff
@@ -35,37 +38,89 @@ const addFileInfo = <T extends NodeError<T>>(error: T): T => {
   if (json !== undefined) {
     const jsonInfo = getInfo(json)
     if (jsonInfo !== undefined) {
-      const path = error.path
-      error.jsonPosition = getDescendantFilePosition(
-        json,
-        path === undefined ? undefined : isArray(path) ? path : path.split("/")
+      const errorPath = error.path
+      assignOptional(
+        error,
+        "jsonPosition",
+        getDescendantFilePosition(
+          json,
+          errorPath === undefined ? undefined :
+            isArray(errorPath) ? errorPath :
+            errorPath.split("/")
+        )
       )
       error.jsonUrl = getRootObjectInfo(jsonInfo).url
     }
   }
   return error
 }
 
+const splitPathAndReverse = (p: string | undefined) =>
+  p === undefined ? undefined : Array.from(flatMap(p.split("/"), s => s.split("\\"))).reverse()
+
+const isSubPath = (mainPath: ReadonlyArray<string> | undefined, subPath: ReadonlyArray<string>) =>
+  mainPath !== undefined &&
+  mainPath.length > subPath.length &&
+  subPath.every((s, i) => mainPath[i] === s)
+
 const createErrorProcessor = <T extends NodeError<T>>(suppression: Suppression | undefined) => {
 
   const isSuppressed = suppression === undefined ?
     () => false :
-    (error: T): boolean =>
-      // TODO: JSONPath: https://www.npmjs.com/package/jsonpath using jp.nodes() function.
+    (error: T): boolean => {
+      const urlReversed = splitPathAndReverse(error.url)
```

Comment:
maybe a good idea to a comment on the fact that the path may be partial and you're trying to see if the path in the suppression is included in the one that comes from the error.